UOP1_Project/Assets/Scripts/StateMachineTest/ScriptableCloseToTargetCondition.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using DeivSky.StateMachine;
+using DeivSky.StateMachine.Scriptables;
+using UnityEngine;
+
+[CreateAssetMenu(fileName = "CloseToTarget", menuName = "State Machines/Tests/Conditions/Close To Target")]
+public class ScriptableCloseToTargetCondition : ScriptableStateCondition
```

Comment:
All throughout the implementation you're using "Scriptable" as a prefix for classes that represent ScriptableObjects. 

But "scriptable" just means that they can contain some behaviour, in this sense MonoBehaviours are scriptable too, and it makes it all very confusing.
I believe it would be better to use the prefix "ScriptableObject", or "SO", to indicate that they are Scriptable Objects.

For instance, this class would be `SOCloseToTargetCondition` or (my preference) `CloseToTargetConditionSO`.

Reply:
I agree and I admit I'm not particularly good with naming. I like SO at the end, I'll start using it!

Reply:
You don't need to bash yourself :D